backend_new/src/services/application-csv-export.service.ts

```diff
@@ -0,0 +1,653 @@
+            val
+              .map((race) => this.convertDemographicRaceToReadable(race))
+              .join(','),
+        },
+        {
+          path: 'demographics.howDidYouHear',
+          label: 'How did you Hear?',
+        },
+      );
+    }
+
+    return headers;
+  }
+
+  addressToString(address: Address): string {
+    return `${address.street}${address.street2 ? ' ' + address.street2 : ''} ${
+      address.city
+    }, ${address.state} ${address.zipCode}`;
+  }
+
+  multiselectQuestionFormat(question: ApplicationMultiselectQuestion): string {
+    if (!question) return '';
+    const address = question.options.reduce((_, curr) => {
+      const extraData = curr.extraData.find((data) => data.type === 'address');
+      return extraData ? extraData.value : '';
+    }, {}) as Address;
+    return this.addressToString(address);
+  }
+
+  convertDemographicRaceToReadable(type: string): string {
+    const [rootKey, customValue = ''] = type.split(':');
+    const typeMap = {
+      americanIndianAlaskanNative: 'American Indian / Alaskan Native',
+      asian: 'Asian',
+      'asian-asianIndian': 'Asian[Asian Indian]',
+      'asian-otherAsian': `Asian[Other Asian:${customValue}]`,
+      blackAfricanAmerican: 'Black / African American',
+      'asian-chinese': 'Asian[Chinese]',
+      declineToRespond: 'Decline to Respond',
+      'asian-filipino': 'Asian[Filipino]',
+      'nativeHawaiianOtherPacificIslander-guamanianOrChamorro':
+        'Native Hawaiian / Other Pacific Islander[Guamanian or Chamorro]',
+      'asian-japanese': 'Asian[Japanese]',
+      'asian-korean': 'Asian[Korean]',
+      'nativeHawaiianOtherPacificIslander-nativeHawaiian':
+        'Native Hawaiian / Other Pacific Islander[Native Hawaiian]',
+      nativeHawaiianOtherPacificIslander:
+        'Native Hawaiian / Other Pacific Islander',
+      otherMultiracial: `Other / Multiracial:${customValue}`,
+      'nativeHawaiianOtherPacificIslander-otherPacificIslander': `Native Hawaiian / Other Pacific Islander[Other Pacific Islander:${customValue}]`,
+      'nativeHawaiianOtherPacificIslander-samoan':
+        'Native Hawaiian / Other Pacific Islander[Samoan]',
+      'asian-vietnamese': 'Asian[Vietnamese]',
+      white: 'White',
+    };
+    return typeMap[rootKey] ?? rootKey;
+  }
+
+  unitTypeToReadable(type: string): string {
+    return typeMap[type] ?? type;
+  }
+
+  private async authorizeCSVExport(user, listingId): Promise<void> {
+    /**
+     * Checking authorization for each application is very expensive.
+     * By making listingId required, we can check if the user has update permissions for the listing, since right now if a user has that
+     * they also can run the export for that listing
+     */
+    /* const jurisdictionId =
+      await this.listingService.getJurisdictionIdByListingId(listingId);
+
+    await this.authzService.canOrThrow(user, 'listing', authzActions.update, {
+      id: listingId,
+      jurisdictionId,
+    }); */
```

Comment:
Does this still need to be commented out?

Reply:
I think since permisioning is in place we should be able to have this function